CMakeLists.txt

```diff
@@ -0,0 +1,357 @@
+#; -*-CMake-*-
+
+# Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.17.0) # decouples C++ and CUDA standards, see https://gitlab.kitware.com/cmake/cmake/issues/19123
+set(ITENSORS_REQUIRED_CMAKE_VERSION 3.17.0 CACHE INTERNAL "")
```

Comment:
Is there a reason you use `ITENSORS` instead of `ITENSOR`? In the C++ version we have only used `ITensor` (not plural), `ITensors` is a bit of a quirk of the Julia version since the name of the package (module) can't be the same as the name of the type inside the package.

Reply:
No I just chose a name. I will change it to ITENSOR

Reply:
@mtfishman Do you know why my CI isn't running? All I did was try to add `apt-get cmake` to the dockerfile

Reply:
I see this on Jenkins:
<img width="1427" alt="Jenkins failure" src="https://github.com/ITensor/ITensor/assets/7855256/182b1b1a-18b2-43ac-82ad-78c464e5ea55">

Looks like a formatting issue, I think those CI workflow files can be finicky about that kind of thing.

I guess you don't have access to see Jenkins? Maybe you can sign up for access by going to https://jenkins.flatironinstitute.org/.

Reply:
Got it, I guess it can also take a minute to get picked up by jenkins too